Tests/SonarScanner.MSBuild.Tasks.IntegrationTests/TargetsTests/RazorTargetTests.cs

```diff
@@ -43,23 +44,67 @@ public void Razor_ForProductProject_CheckErrorLogProperties()
+  <ProjectSpecificOutDir>{sourceDirectory}</ProjectSpecificOutDir>
 </PropertyGroup>
 ";
 
-            string afterTargets = string.Join(";", TargetConstants.SetRazorCodeAnalysisPropertiesTarget);
+            var afterTargets = string.Join(";", TargetConstants.SetRazorCodeAnalysisPropertiesTarget);
 
             var filePath = CreateProjectFile(null, projectSnippet, afterTargets);
 
             // Act
-            var result = BuildRunner.BuildTargets(TestContext, filePath,
-                TargetConstants.SetRazorCodeAnalysisPropertiesTarget);
+            var result = BuildRunner.BuildTargets(TestContext, filePath, TargetConstants.SetRazorCodeAnalysisPropertiesTarget);
 
             // Assert
             result.AssertTargetExecuted(TargetConstants.SetRazorCodeAnalysisPropertiesTarget);
             AssertErrorLogIsSetBySonarQubeTargets(result);
+            File.Exists(Path.Combine(destinationDirectory, "ProtoBuf.txt")).Should().BeTrue();
+        }
+
+        [TestMethod]
+        public void Razor_RazorSpecificProtoBufsAndProjectInfo_AreCopiedToCorrectFolders()
+        {
+            // Arrange
+            var root = TestUtils.CreateTestSpecificFolder(TestContext);
+            var projectSpecificOutDir = Path.Combine(root, "0");
+            var temporaryProjectSpecificOutDir = Path.Combine(root, "0.tmp");
+            var razorSpecificOutDir = Path.Combine(root, "0.razor");
+            Directory.CreateDirectory(projectSpecificOutDir);
+            Directory.CreateDirectory(temporaryProjectSpecificOutDir);
+            File.WriteAllBytes(Path.Combine(projectSpecificOutDir, "ProjectInfo.xml"), new byte[] { });
+            File.WriteAllBytes(Path.Combine(temporaryProjectSpecificOutDir, "ProtoBuf.txt"), new byte[] { });
+
+            var projectSnippet = $@"
+<PropertyGroup>
+  <TemporaryProjectSpecificOutDir>{temporaryProjectSpecificOutDir}</TemporaryProjectSpecificOutDir>
+  <ProjectSpecificOutDir>{projectSpecificOutDir}</ProjectSpecificOutDir>
+</PropertyGroup>
+";
+
+            var afterTargets = string.Join(";", TargetConstants.CopyProjectInfoFileInPresenceOfRazorTarget);
+
+            var filePath = CreateProjectFile(null, projectSnippet, afterTargets);
+
+            // Act
+            var result = BuildRunner.BuildTargets(TestContext, filePath, TargetConstants.CopyProjectInfoFileInPresenceOfRazorTarget);
+
+            // Assert
+            result.AssertTargetExecuted(TargetConstants.CopyProjectInfoFileInPresenceOfRazorTarget);
+            File.Exists(Path.Combine(projectSpecificOutDir, "ProtoBuf.txt")).Should().BeTrue();
+            File.Exists(Path.Combine(projectSpecificOutDir, "ProtoBuf.txt")).Should().BeTrue();
+            File.Exists(Path.Combine(razorSpecificOutDir, "ProjectInfo.xml")).Should().BeTrue();
+            File.Exists(Path.Combine(razorSpecificOutDir, "ProjectInfo.xml")).Should().BeTrue();
+            Directory.Exists(temporaryProjectSpecificOutDir).Should().BeFalse();
```

Comment:
`Razor_ExcludedProject_NoErrorLog` below should also have `<RazorCompileOnBuild>true</RazorCompileOnBuild>` because the entry condition for the test has changed and it works for another reason.